spec/unit/puppet/provider/rabbitmq_user/rabbitmqctl_spec.rb

```diff
@@ -3,213 +3,174 @@
   end
 
-  it 'clears tags while keep admin tag' do
-    resource[:admin] = true
-    provider.expects(:rabbitmqctl).with('-q', 'list_users').returns <<-EOT
-one [administrator]
-foo [administrator, tag1, tag2]
-icinga  [monitoring]
-kitchen []
-kitchen2        [abc, def, ghi]
-EOT
-    provider.expects(:rabbitmqctl).with('set_user_tags', 'foo', ['administrator'])
-    provider.tags = []
-  end
+  describe '#create' do
+    it 'adds a user' do
+      provider.expects(:rabbitmqctl).with('add_user', 'rmq_x', 'secret')
+      provider.create
+    end
+    context 'no password supplied' do
+      let(:resource) do
+        Puppet::Type.type(:rabbitmq_user).new(
+          ensure: :present,
+          name: 'rmq_x'
+        )
+      end
 
-  it 'changes tags while keep admin tag' do
-    resource[:admin] = true
-    provider.expects(:rabbitmqctl).with('-q', 'list_users').returns <<-EOT
-one [administrator]
-foo [administrator, tag1, tag2]
-icinga  [monitoring]
-kitchen []
-kitchen2        [abc, def, ghi]
-EOT
-    provider.expects(:rabbitmqctl).with('set_user_tags', 'foo', %w[administrator tag1 tag3 tag7])
-    provider.tags = %w[tag1 tag7 tag3]
+      it 'raises an error' do
+        expect do
+          provider.create
+        end.to raise_error(Puppet::Error, 'Password is a required parameter for rabbitmq_user (user: rmq_x)')
+      end
+    end
   end
 
-  it 'creates user with tags and without admin' do
-    resource[:tags] = %w[tag1 tag2]
-    provider.expects(:rabbitmqctl).with('add_user', 'foo', 'bar')
-    provider.expects(:rabbitmqctl).with('set_user_tags', 'foo', %w[tag1 tag2])
-    provider.expects(:rabbitmqctl).with('-q', 'list_users').returns <<-EOT
-foo []
-EOT
-    provider.create
+  describe '#destroy' do
+    it 'removes a user' do
+      provider.expects(:rabbitmqctl).with('delete_user', 'rmq_x')
+      provider.destroy
+    end
   end
 
-  it 'creates user with tags and with admin' do
-    resource[:tags] = %w[tag1 tag2]
-    resource[:admin] = true
-    provider.expects(:rabbitmqctl).with('add_user', 'foo', 'bar')
-    provider.expects(:rabbitmqctl).with('-q', 'list_users').twice.returns <<-EOT
-foo []
-EOT
-    provider.expects(:rabbitmqctl).with('set_user_tags', 'foo', ['administrator'])
-    provider.expects(:rabbitmqctl).with('set_user_tags', 'foo', %w[administrator tag1 tag2])
-    provider.create
+  describe '#check_password' do
+    context 'correct password' do
+      before do
+        provider.class.stubs(:rabbitmqctl).with(
+          'eval',
+          'rabbit_access_control:check_user_pass_login(list_to_binary("rmq_x"), list_to_binary("secret")).'
+        ).returns <<-EOT
+{ok,{user,<<"rmq_x">>,[],rabbit_auth_backend_internal,
+          {internal_user,<<"rmq_x">>,
+                         <<193,81,62,182,129,135,196,89,148,87,227,48,86,2,154,
+                           192,52,119,214,177>>,
+                         []}}}
+EOT
+      end
+
+      it do
+        provider.check_password('secret')
```

Comment:
I don't think this is doing anything useful, but need to come up with a better test here.